llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -494,6 +499,148 @@ class DefaultStrategy : public PostPipelinerStrategy {
   }
 };
 
+class ConfigStrategy : public PostPipelinerStrategy {
+public:
+  // None serves as a terminator; any component after it will be ignored.
+  enum PriorityComponent {
+    None,
+    NodeNum,
+    Latest,
+    Critical,
+    Sibling,
+    LCDLatest
+  };
+  static const int Shift = 3;
+  static const int Size = 1 << Shift;
+  static const int Mask = Size - 1;
+
+private:
+  const std::string Name;
+  std::set<int> SuccSiblingScheduled;
+  std::function<bool(const SUnit &A, const SUnit &B)> Discriminators[Size] = {
```

Comment:
Nit: Might be more readable to define all those functions separately, and then have e.g.
```
std::function<bool(const SUnit &A, const SUnit &B)> Discriminators[Size] = {
  &CompareNone,
  &CompareNodeNum,
  &CompareLatest,
  ...
};
```

Reply:
They would have to be methods, since they capture e.g. Info.

Reply:
Or just named lambdas within the constructor or wherever we fill in `Discriminators`. Anyway, it's just a nit, but I feel it will become useful to have names when we start to have dozens of compare functions.